src/components/documentManager.ts

```diff
@@ -0,0 +1,611 @@
+     *
+     * @param {BaseItemDto} item to look up
+     * @returns {string} html to put in document
+     */
+    private static getRatingHtml(item: BaseItemDto): string {
+        let html = '';
+        if (item.CommunityRating != null) {
+            html +=
+                `<div class="starRating" title="${item.CommunityRating}"></div>` +
+                '<div class="starRatingValue">' +
+                item.CommunityRating.toFixed(1) +
+                '</div>';
+        }
+
+        if (item.CriticRating != null) {
+            const verdict = item.CriticRating >= 60 ? 'fresh' : 'rotten';
+            html +=
+                `<div class="${verdict} rottentomatoesicon" title="${verdict}"></div>` +
+                `<div class="criticRating">${item.CriticRating}%</div>`;
+        }
+
+        return html;
+    }
+
+    /**
+     * Set the status of the app, and switch the visible view
+     * to the corresponding one.
+     *
+     * @param {string} status to set
+     */
+    public static setAppStatus(status: string): void {
+        this.status = status;
+        document.body.className = status;
+    }
+
+    /**
+     * Get the status of the app
+     *
+     * @returns {string} app status
+     */
+    public static getAppStatus(): string {
+        return this.status;
+    }
+
+    /**
+     * BACKDROP LOGIC
+     *
+     * Backdrops are set on the waiting container.
+     * They are switched around every 30 seconds by default
+     * (governed by startBackdropInterval)
+     *
+     * @param {BaseItemDto | null} item Item to use for waiting backdrop, null to remove it.
+     */
+    public static setWaitingBackdrop(item: BaseItemDto | null): void {
+        // no backdrop as a fallback
+        let src: string | null = null;
+
+        if (item != null) {
+            if (item.BackdropImageTags && item.BackdropImageTags.length) {
+                // get first backdrop of image if applicable
+                src = JellyfinApi.createUrl(
+                    `Items/${item.Id}/Images/Backdrop/0?tag=${item.BackdropImageTags[0]}`
+                );
+            } else if (
+                item.ParentBackdropItemId &&
+                item.ParentBackdropImageTags &&
+                item.ParentBackdropImageTags.length
+            ) {
+                // otherwise get first backdrop from parent
+                src = JellyfinApi.createUrl(
+                    `Items/${item.ParentBackdropItemId}/Images/Backdrop/0?tag=${item.ParentBackdropImageTags[0]}`
+                );
+            }
+        }
+
+        const element: HTMLElement | null = document.querySelector(
+            '#waiting-container-backdrop'
+        );
+
+        if (element === null) {
+            console.error(
+                'documentManager: Cannot find #waiting-container-backdrop'
+            );
+        } else {
+            (<HTMLElement>element).style.backgroundImage =
+                src != null ? `url(${src})` : '';
```

Comment:
I don't really like this because the new image loading is visible. I would prefer to preload the new src and then do something like a buffer swap. Someone better at frontend stuff that could help me with this?

Reply:
@hawken93 I think this is what you want:

https://github.com/jellyfin/jellyfin-vue/pull/408/files#diff-4d53a82f6d48248fa1a22bd2f449b0604526d455c84d8b528ab723182b8f4c76R20

Reply:
Cool! I'll see what I can do

Reply:
Right. Ty :+1: